pkg/athena/datasource.go

```diff
@@ -159,3 +160,20 @@ func (s *AthenaDatasource) Workgroups(ctx context.Context, options sqlds.Options
 	}
 	return api.Workgroups(ctx)
 }
+
+func parseArgs(queryArgs json.RawMessage) (sqlds.Options, error) {
+	args := map[string]interface{}{}
+	if queryArgs != nil {
+		err := json.Unmarshal(queryArgs, &args)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse query args: %w", err)
+		}
+	}
+
+	options := sqlds.Options{}
+	for k, v := range args {
+		options[k] = fmt.Sprintf("%v", v)
```

Comment:
Is it fine to convert `v` to a string in this way? I was previously doing this https://github.com/grafana/athena-datasource/pull/214/commits/7fdf12b664b06bd803fd8474c34fbfa52b953729.

Reply:
In that commit you linked, was there a reason to get rid of the `toString` function?

So basically we're never going to know what we're unmarshalling? There is nothing we can describe about `args`'s types? We can only say it's interface{}?

My thought, but I ought to look at the rest of this PR and the part in grafana-aws-sdk to understand what will come out: %v is the most general string formatting verb, might not be in the format that the downstream consumers expect. https://gobyexample.com/string-formatting

Reply:
I don't think it's strictly wrong, but it's probably better practice to do the toString function with a switch statement for every type we expect and throw an error for unexpected types.